Many test failures in dom0 due to recent sd-app changes

## Steps to reproduce

Run `make test` in your `dom0` checkout with a recently provisioned SecureDrop Workstation using latest nightlies.

## Expected behavior

Tests pass

## Actual behavior

Lots of test failures, mainly due to the string "Setting /usr/bin/securedrop-client to enforce mode" appearing in command output. Example: 

```
FAIL: test_mimeapps (test_app.SD_App_Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/erik/securedrop-workstation/tests/test_app.py", line 30, in test_mimeapps
    self.assertEqual(results, expected_results)
AssertionError: 'Setting /usr/bin/securedrop-client to enf[46 chars]top;' != '2 \n    295 open-in-dvm.desktop;'
+ 2 
- Setting /usr/bin/securedrop-client to enforce mode.
-       2 
      295 open-in-dvm.desktop;
```


## Explanation

Nightlies broke on October 5 and were just fixed today, so we did not notice this earlier. https://github.com/freedomofpress/securedrop-client/pull/1153 changed the logic for updating the `QUBES_GPG_DOMAIN` to also invoke `sudo aa-enforce /usr/bin/securedrop-client`. That command produces the aforementioned output. Because this is done in `profile.d`, the command runs on every `qvm-run sd-app` command from `dom0`, which breaks the `sd-app` tests.

We can redirect the `aa-enforce` output to `/dev/null`, which would fix most of the tests, but that assumes we really want to run `aa-enforce` this frequently. In any event, [this test](https://github.com/freedomofpress/securedrop-workstation/blob/main/tests/test_app.py#L12-L15) directly reads the contents of the file that's been modified as part of https://github.com/freedomofpress/securedrop-client/pull/1153, and will still need to be updated.

--- a/tests/base.py
+++ b/tests/base.py
@@ -125,3 +125,27 @@ remotevm = sd-log
         # so let's investigate further.
         # cmd_output = self._run("sudo grep -F \"action 'action-0-omprog' suspended (module 'omprog')\" /var/log/syslog | wc -l").strip()  # noqa
         # self.assertTrue(cmd_output == "0")
+
+    def qubes_gpg_domain_configured(self, vmname=False):
+        """
+        Ensure the QUBES_GPG_DOMAIN is properly set for a given AppVM. This
+        var is set by a script /etc/profile.d.
+        sd-app should have it set to sd-gpg.
+        All other AppVMs should not have this configured.
+        """
+        expected_profile_app = 'if [ "$(qubesdb-read /name)" = "sd-app" ]; then export QUBES_GPG_DOMAIN="sd-gpg"; fi\n'  # noqa: E501
+        expected_env_app = "sd-gpg"
+        expected_env_all = ""
+        profile_path = "/etc/profile.d/sd-app-qubes-gpg-domain.sh"
+
+        env_cmd = 'echo "$QUBES_GPG_DOMAIN"'
+        env_contents = self._run(env_cmd)
+
+        if vmname == "sd-app":
+            self.assertTrue(self._fileExists(profile_path))
+            profile_contents = self._get_file_contents(profile_path)
+            self.assertEqual(profile_contents, expected_profile_app)
+            self.assertEqual(env_contents, expected_env_app)
+        else:
+            self.assertFalse(self._fileExists(profile_path))
+            self.assertEqual(env_contents, expected_env_all)

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -9,10 +9,8 @@ class SD_App_Tests(SD_VM_Local_Test):
         self.vm_name = "sd-app"
         super(SD_App_Tests, self).setUp()
 
-    def test_decrypt_sd_user_profile(self):
-        contents = self._get_file_contents("/etc/profile.d/sd-app-qubes-gpg-domain.sh")
-        expected_content = 'export QUBES_GPG_DOMAIN="sd-gpg"\n'
-        self.assertEqual(contents, expected_content)
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
 
     def test_open_in_dvm_desktop(self):
         contents = self._get_file_contents("/usr/share/applications/open-in-dvm.desktop")

--- a/tests/test_gpg.py
+++ b/tests/test_gpg.py
@@ -52,6 +52,9 @@ class SD_GPG_Tests(SD_VM_Local_Test):
         # Logging to sd-log should be disabled on sd-gpg
         self.assertFalse(self._fileExists("/etc/rsyslog.d/sdlog.conf"))
 
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
+
 
 def load_tests(loader, tests, pattern):
     suite = unittest.TestLoader().loadTestsFromTestCase(SD_GPG_Tests)

--- a/tests/test_log_vm.py
+++ b/tests/test_log_vm.py
@@ -49,6 +49,9 @@ class SD_Log_Tests(SD_VM_Local_Test):
         # Confirm we don't have 'host' entries from Whonix VMs
         self.assertFalse("host" in log_dirs)
 
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
+
 
 def load_tests(loader, tests, pattern):
     suite = unittest.TestLoader().loadTestsFromTestCase(SD_Log_Tests)

--- a/tests/test_proxy_vm.py
+++ b/tests/test_proxy_vm.py
@@ -52,6 +52,9 @@ class SD_Proxy_Tests(SD_VM_Local_Test):
                     actual_app = self._run("xdg-mime query default {}".format(mime_type))
                     self.assertEqual(actual_app, expected_app)
 
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
+
 
 def load_tests(loader, tests, pattern):
     suite = unittest.TestLoader().loadTestsFromTestCase(SD_Proxy_Tests)

--- a/tests/test_sd_devices.py
+++ b/tests/test_sd_devices.py
@@ -45,6 +45,9 @@ class SD_Devices_Tests(SD_VM_Local_Test):
         for line in expected_contents:
             self.assertTrue(line in contents)
 
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
+
 
 def load_tests(loader, tests, pattern):
     suite = unittest.TestLoader().loadTestsFromTestCase(SD_Devices_Tests)

--- a/tests/test_sd_whonix.py
+++ b/tests/test_sd_whonix.py
@@ -74,6 +74,9 @@ class SD_Whonix_Tests(SD_VM_Local_Test):
             "Whonix GW torrc contains duplicate %include lines",
         )
 
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
+
 
 def load_tests(loader, tests, pattern):
     suite = unittest.TestLoader().loadTestsFromTestCase(SD_Whonix_Tests)

--- a/tests/test_viewer.py
+++ b/tests/test_viewer.py
@@ -36,6 +36,9 @@ class SD_Viewer_Tests(SD_VM_Local_Test):
                     actual_app = self._run("xdg-mime query default {}".format(mime_type))
                     self.assertEqual(actual_app, expected_app)
 
+    def test_gpg_domain_configured(self):
+        self.qubes_gpg_domain_configured(self.vm_name)
+
 
 def load_tests(loader, tests, pattern):
     suite = unittest.TestLoader().loadTestsFromTestCase(SD_Viewer_Tests)

